src/BlockStorage/v2/Models/Snapshot.php

```diff
@@ -49,7 +49,7 @@ class Snapshot extends OperatorResource implements Listable, Creatable, Updateab
     protected $markerKey = 'id';
 
     protected $aliases = [
-        'created_at' => 'createdAt',
+        'created_at' => ['createdAt', \DateTimeImmutable::class],
```

Comment:
so I'm thinking of the best way we can both represent the data and abstract the parsing functionality.

instead of using a flat array (which imparts no structure), how about we create a new `Common\Resource\Alias` class? we could then move all of this alias logic inside that class and clean up `common\Resource\AbstractResource` a bit.

Reply:
If I understood correctly, `protected $aliases` will be an instance of `Common\Resource\Alias` object.

In this case initialization method should be added where alias object could be configured. Ideally if Alias object would be a static member, I don't see any reason initialize it upon each object construction.

As a variant `AbstractResource` can contain protected static method `getAlias` that should return an instance of `Common\Resource\Alias` class (here an object can be static private member, or configure parent's object, or whatever you want). This method will be used then inside of `AbstractResource::populateFromArray` method (e.g. `static::getAlias()->setProperty($this, $key, $value)`).

Reply:
I was thinking more like:

```php
protected $aliases = [
   'created_at' => new Alias('createdAt', \DateTimeImmutable::class),
   'foo'        => new Alias('foo'),
];
```

```php
public function populateFromArray(array $array): self
{
   foreach ($array as $key => $val) {
      if isset($this->aliases[$key]) {
         $alias = $this->aliases[$key];
         $key = $alias->getKey();
         $val = $alias->getVal($val);
      }
      $this->$key = $val;
   }
}
```

and move all the casting logic to `Alias::getVal`.

Reply:
There are two problems:
```php 
protected $aliases = [
   'created_at' => new Alias('createdAt', \DateTimeImmutable::class),
   'foo'        => new Alias('foo'),
];
```
this won't work, expressions are not allowed for default values as far as I know. Or did I miss something?

But even so this is additional object constructions (that can be avoided) upon each `populateFromArray` call.